src/game/Spell.cpp

```diff
@@ -3371,6 +3371,13 @@ void Spell::SendLogExecute()
         {
             switch (m_spellInfo->Effect[EFFECT_INDEX_0])
             {
+                case SPELL_EFFECT_DISPEL:
+                case SPELL_EFFECT_THREAT:
```

Comment:
You may want to try and move these two cases down here: https://github.com/cmangos/mangos-classic/blob/0ed121835e5b967f4976ec2d20531529be5243f0/src/game/Spell.cpp#L3452 because then you should be able to remove the functionality you added in the following lines, as those cases by the link go through basically the same.